File: src/main/java/com/github/easylive1989/flutterdevelopmenttool/PopupDialogAction.java
package com.github.easylive1989.flutterdevelopmenttool;

import com.intellij.ide.DataManager;
import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.CommonDataKeys;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.Messages;
import com.intellij.psi.*;
import com.intellij.psi.util.PsiTreeUtil;
import com.jetbrains.lang.dart.psi.DartMethodDeclaration;
import org.jetbrains.annotations.NotNull;

public class PopupDialogAction extends AnAction {
    @Override
    public void actionPerformed(@NotNull AnActionEvent event) {
        int offset = CommonDataKeys.CARET.getData(DataManager.getInstance().getDataContext()).getOffset();
        PsiFile psiFile = event.getData(CommonDataKeys.PSI_FILE);
        var message = "";
        if (psiFile == null) {
            show(event, message);
            return;
        }

        PsiElement element = psiFile.findElementAt(offset);
        if (element == null) {
            show(event, message);
            return;
        }

        message += String.format("\nSelected Element: %s", element);
        message += String.format("\nSelected Language: %s", element.getLanguage());

        DartMethodDeclaration method = PsiTreeUtil.getParentOfType(element, DartMethodDeclaration.class);
        if (method == null) {
            show(event, message);
            return;
        }

//      Query<PsiReference> result = MethodReferencesSearch.search(containingMethod);
        message += String.format("\nSelected Method: %s", method);

        show(event, message);
    }

    private void show(AnActionEvent event, String message) {
        Project currentProject = event.getProject();
        StringBuffer dlgMsg = new StringBuffer(event.getPresentation().getText() + " Selected!");
        String dlgTitle = event.getPresentation().getDescription();
        dlgMsg.append(message);
        Messages.showMessageDialog(currentProject, dlgMsg.toString(), dlgTitle, Messages.getInformationIcon());
    }
}

File: src/main/kotlin/com/github/easylive1989/flutterdevelopmenttool/actions/PopupDialogAction.kt
package com.github.easylive1989.flutterdevelopmenttool.actions

import com.intellij.ide.DataManager
import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.actionSystem.CommonDataKeys
import com.intellij.openapi.project.Project
import com.intellij.openapi.ui.Messages
import com.intellij.psi.PsiClass
import com.intellij.psi.PsiFile
import com.intellij.psi.PsiMethod
import com.intellij.psi.search.searches.MethodReferencesSearch
import com.intellij.psi.util.PsiTreeUtil
import org.jetbrains.annotations.NotNull


class PopupDialogAction : AnAction() {

    override fun update(e: AnActionEvent) {
        // Set the availability based on whether a project is open
        // Set the availability based on whether a project is open
        val project = e.project
        e.presentation.isEnabledAndVisible = project != null
    }

    override fun actionPerformed(@NotNull event: AnActionEvent) {
        val currentProject: Project? = event.getProject()
        val dlgMsg = StringBuffer(event.getPresentation().getText().toString() + " Selected!")
        val dlgTitle: String = event.getPresentation().getDescription()
        val offset: Int?= CommonDataKeys.CARET.getData(DataManager.getInstance().getDataContext())?.offset;
        val psiFile: PsiFile? = event.getData(CommonDataKeys.PSI_FILE)
        val element = psiFile?.findElementAt(offset!!)
        var message = ""
        if (element != null) {
            message += java.lang.String.format("\nSelected Element: %s", element.toString())
            val containingMethod: PsiMethod? = PsiTreeUtil.getParentOfType(element, PsiMethod::class.java)
            if (containingMethod != null) {
                val result = MethodReferencesSearch.search(containingMethod)
                message += java.lang.String.format("\nSelected Method: %s", containingMethod.firstChild.language.toString())
                val containingClass: PsiClass? = containingMethod.getContainingClass()
                if (containingClass != null) {
                    message += java.lang.String.format("\nSelected Class: %s", containingClass.toString())
                }
            }
        }

        dlgMsg.append(message)
        Messages.showMessageDialog(currentProject, dlgMsg.toString(), dlgTitle, Messages.getInformationIcon())
    }
}
